conan/tools/files/files.py

```diff
@@ -367,11 +351,10 @@ def print_progress(_, __):
                     output.error(f"Error extract {file_.filename}\n{str(e)}", error_type="exception")
         output.writeln("")
 
-
-def untargz(filename, destination=".", pattern=None, strip_root=False, extract_filter=None):
+def untargz(filename, destination=".", pattern=None, strip_root=False, extract_filter=None, output=None):
     # NOT EXPOSED at `conan.tools.files` but used in tests
     import tarfile
-    with tarfile.TarFile.open(filename, 'r:*') as tarredgzippedFile:
+    with tarfile.TarFile.open(fileobj=FileProgress(filename, msg="Uncompressing"), mode='r:*') as tarredgzippedFile:
```

Comment:
This ``fileob`` is not being closed correctly, as it is not called with ``with FileProgres``. It would be better if this file is being correctly closed

Reply:
Using now `with` statement to manage both file and resource.